tests/trx_generator/trx_provider.hpp

```diff
@@ -85,17 +105,40 @@ namespace eosio::testing {
    };
 
    struct http_connection : public provider_connection {
+      std::mutex                                                     _trx_info_map_lock;
+      std::map<eosio::chain::transaction_id_type, ackedTrxTraceInfo> _acked_trx_trace_info_map;
+
       std::atomic<uint64_t> _acknowledged{0};
       std::atomic<uint64_t> _sent{0};
 
       explicit http_connection(const provider_base_config& provider_config)
           : provider_connection(provider_config) {}
 
       void send_transaction(const chain::packed_transaction& trx) final;
+      void record_trx_info(eosio::chain::transaction_id_type trx_id, unsigned int block_num, unsigned int cpu_usage_us,
+                           unsigned int net_usage_words, const std::string& block_time) {
+         std::lock_guard<std::mutex> lock(_trx_info_map_lock);
+         _acked_trx_trace_info_map.insert({trx_id, {true, block_num, cpu_usage_us, net_usage_words, block_time}});
+      }
+
+      ackedTrxTraceInfo get_acked_trx_trace_info(const eosio::chain::transaction_id_type& trx_id) {
+         ackedTrxTraceInfo           info;
+         std::lock_guard<std::mutex> lock(_trx_info_map_lock);
+         auto                        search = _acked_trx_trace_info_map.find(trx_id);
+         if (search != _acked_trx_trace_info_map.end()) {
+            info = search->second;
+         } else {
+            elog("get_acked_trx_trace_info - Acknowledged transaction trace info not found for transaction with id: "
+                 "${id}",
```

Comment:
Please make this one line instead of 2.

Reply:
Addressed: [f9838a1](https://github.com/AntelopeIO/leap/pull/1229/commits/f9838a118b4018c23a0b113543aa17ef0e116412)